src/core/pile.sol

```diff
@@ -128,11 +128,18 @@ contract Pile is DSNote, TitleOwned {
         beans.drip(loans_[loan].fee);
     }
 
-    // repay() a certain amount of token from the user to the Pile
-    function repay(uint loan, uint wad) public owner(loan) note {
-        // moves currency from usr to pile and reduces debt
-        require(loans_[loan].balance == 0,"before repay loan needs to be withdrawn");
+
+    // recovery used for defaulted loans_
+    function recovery(uint loan, uint wad) public auth {
+        doRepay(loan, wad);
+
+        uint loss = beans.debtOf(loan, loans_[loan].fee);
+        beans.decLoanDebt(loan, loans_[loan].fee, loss);
+    }
+
+    function doRepay(uint loan, uint wad) internal {
         collect(loan);
+
         uint fee = loans_[loan].fee;
         uint debt = beans.debtOf(loan, fee);
 
```

Comment:
@manuelpolzhofer I think this check below prevents partial repay and consequently also recovery (which might be a partial repay)